### DISCUSSION TITLE
Setting a Product ID or SKU

### DISCUSSION THREAD

**Discussion by joybcode** [duncanmcclean/simple-commerce#1130] [Help]
Hi,

I'm new to to statamic and simple-commerce. I've purchased your addon.
Switching over from Wordpress.

How do I set a unique product ID or sku number or does this get generated automatically?  Please see screenshot from our current wordpress site.

![image](https://github.com/user-attachments/assets/e578ffde-4951-4c79-87b4-8f63bd9f631c)

Many thanks

**Comment by duncanmcclean** [comment]
When products are created, Statamic assigns them a unique ID like `52bab544-d29e-40cb-a7c3-1716190fd19f`.

If you need something shorter, then you'd need to generate it yourself. You could maybe achive it by listening to Statamic's [`EntrySaving`](https://statamic.dev/extending/events#entrysaving) event.

It's not something that Simple Commerce includes out of the box.

**Reply by joybcode** [reply]
Thanks for a very quick response.

**Comment by sstottelaar** [comment]
Every product is an entry with a unique ID like "f0461794-6f22-4be8-b003-4514daa6123a". As for an SKU, you can easily add a custom field called 'SKU'. This may be auto-generated in WordPress based on the title. If so, this is something you have to implement yourself.

**Reply by joybcode** [reply]
Thanks for you help. I'll try this method.
